### PR TITLE
Move GCI qa/ci jobs into their own tab

### PR DESCRIPTION
**Pull request by wonderfly**
The k8s-e2e-gci-[ci|qa]-* jobs are used for GCI's development, and will
be of little interest to k8s developers. Having them under a separate
dashboard tab reduces confusion.

### CODE REVIEW

**Comment by wonderfly** [comment]
@krzyzacy I'll merge this one once @adityakali takes a look

**Comment by wonderfly** [comment]
@krzyzacy The //jenkins:bootstrap_test fails for me even at HEAD. Is it flaky now? Or did I do something wrong?

File: testgrid/config/config.yaml [hunk 1]
```diff
@@ -1039,10 +1039,12 @@ dashboards:
   - name: gce-ha-master
     test_group_name: kubernetes-e2e-gce-ha-master
 
-- name: google-gci
+# "google-gci-dev" is a private tab that should be looked at by GCI/COS
```
Comment (adityakali) [inline testgrid/config/config.yaml]: I suggest rewording this as this is not 'private' by any means (permissions or anything :) ).
May be also add that when making changes here, please cc @kubernetes/goog-image .

File: testgrid/config/config.yaml [hunk 2]
```diff
@@ -1039,10 +1039,12 @@ dashboards:
...
+# developers. Tests under this tab are used for GCI/COS image development.
```
Comment (adityakali) [inline testgrid/config/config.yaml]: s/image development/image qualification/ ?

File: testgrid/config/config.yaml [hunk 3]
```diff
@@ -1039,10 +1039,12 @@ dashboards:
...
+# Kubernetes developers interested in k8s on GCI test results should look at the
+# "google-gci" tab instead.
+- name: google-gci-dev
```
Comment (adityakali) [inline testgrid/config/config.yaml]: name this 'google-gci-qa' to be more clear?

File: testgrid/config/config.yaml [hunk 4]
```diff
@@ -1099,6 +1099,14 @@ dashboards:
     test_group_name: kubernetes-e2e-gce-gci-qa-slow-m57
   - name: qa-slow-master
     test_group_name: kubernetes-e2e-gce-gci-qa-slow-master
+# END google-gci-dev
+
+- name: google-gci
```
Comment (adityakali) [inline testgrid/config/config.yaml]: This needs similar explanation too. May be mention that any GCI/COS specific tests that are not part of COS release qualification can be added here?

**Comment by spxtr** [comment]
@wonderfly fixed in #1924.

**Comment by krzyzacy** [comment]
@k8s-bot bazel test this

File: testgrid/config/config.yaml (hunk 3 above)
Comment (krzyzacy) [inline testgrid/config/config.yaml]: +1 since all the test names are -gci-qa-

File: testgrid/config/config.yaml (hunk 1 above)
Comment (wonderfly) [inline testgrid/config/config.yaml]: Done

File: testgrid/config/config.yaml (hunk 2 above)
Comment (wonderfly) [inline testgrid/config/config.yaml]: I still like to include "development" because the gci-ci-* jobs are to qualify a newer k8s version before it's builtin to GCI, as opposed to qualifying a GCI image before release. So I made it "development AND qualification".

File: testgrid/config/config.yaml (hunk 3 above)
Comment (wonderfly) [inline testgrid/config/config.yaml]: Done

Comment (wonderfly) [inline testgrid/config/config.yaml]: As explained above. I think "dev" is more inclusive because it also includes the "gci-ci-*" tests.

File: testgrid/config/config.yaml (hunk 4 above)
Comment (wonderfly) [inline testgrid/config/config.yaml]: I added some brief explanation in the comments for the google-gci-dev tab above. I don't want to sound like I am _defining_ what this tab should be used for, especially that I am not entirely sure what people might use it for.

**Comment by wonderfly** [comment]
@adityakali @krzyzacy PTAL

**Comment by krzyzacy** [comment]
LG for me, give @adityakali a chance to take another look

**Comment by wonderfly** [comment]
cc/ @kewu1992 Would this affect our release dashboard?

**Comment by kewu1992** [comment]
@wonderfly Unlikely, release dashboard queries by test group name (e.g. ci-kubernetes-e2e-gce-gci-qa-master) regardless of dashboard tab.

File: testgrid/config/config.yaml [hunk 5]
```diff
@@ -1039,10 +1039,12 @@ dashboards:
   - name: gce-ha-master
     test_group_name: kubernetes-e2e-gce-ha-master
 
-- name: google-gci
+# "google-gci-dev" is a tab for tests that are used for GCI(aka COS) image
+# development and qualification. It should not be confused with the "google-gci"
+# tab that is for k8s development and release qualification.
+# If you are going to make a change here, kindly cc @kubernetes/goog-image.
+- name: google-gci-dev
```
Comment (adityakali) [inline testgrid/config/config.yaml]: The idea behind suggesting 'google-gci-qa' name was to indicate the _purpose_ of the tests, rather than what tests it contains or what they are named. But this is a moot point.
